docs/architecture/accounts.md

```diff
@@ -183,19 +182,23 @@ A user can create an account in one of the following manners:
 
 ## Account types
 
-There are four types of accounts in Miden:
+There are two basic types of accounts in Miden. Regular accounts and faucets. Only faucets can mint new assets. Regular accounts can be mutable or immutable, which simply means that the account code can change after creation or not. 
 
-| | Regular updatable account | Regular immutable account | Faucet for fungible assets | Faucet for non-fungible assets |
+Type and mutability is encoded in the most significant bits of the account's ID. 
+
+| | Basic mutable | Basic immutable | Fungible faucet | Non-fungible faucet |
 |---|---|---|---|---|
 | **Description** | For most users, e.g. a wallet. Code changes allowed, including public API. | For most smart contracts. Once deployed code is immutable. | Users can issue fungible assets and customize them. | Users can issue non-fungible assets and customize them. |
 | **Code updatability** | yes | no | no | no |
 | **Most significant bits** | `00` | `01` | `10` | `11` |
 
-## Account storage modes
+## Public and private accounts
+
+Users can decide whether to keep their accounts private or public at account creation. The account ID encodes the preference on the third and fourth most significant bit.
 
-* Accounts with **public state**, where the actual state is stored on-chain. These would be similar to how accounts work in public blockchains. Smart contracts that depend on public shared state should be stored public on Miden, e.g., DEX contract.
-* Accounts with **private state**, where only the hash of the account is stored on-chain. Users who want stay private and take care of their own data should choose this mode. The hash is defined as: `hash([account ID, 0, 0, nonce], [vault root], [storage root], [code root])`.
+* Accounts with **public state**: the actual state is stored on-chain. These would be similar to how accounts work in public blockchains, like Ethereum. Smart contracts that depend on public shared state should be stored public on Miden, e.g., DEX contract.
+* Accounts with **private state**: only the hash of the account is stored on-chain. Users who want stay private and take care of their own data should choose this mode. Users who want to interact with private accounts need to know the account's interface.
 
-In the future we will also support **encrypted state** which will be on-chain but encrypted. * Depending on the account storage mode (private vs. encrypted vs. public) and transaction type (local vs. network) the operator receives the new Account ID eventually and - if the transaction is correct - adds the ID to the Account DB
+In the future we will also support **encrypted state** which will be on-chain but encrypted. 
```

Comment:
I would avoid any statements about the future which become problematic in technical docs. What tends to happen if you allow these is docs become full of statements about the future, which often don't transpire, and it's very difficult to keep track of them and update them. Better to leave this for blog posts and marketing material.

Reply:
good point